snapcraft/internal/sources/_base.py

```diff
@@ -62,3 +65,35 @@ def download(self):
             request.raise_for_status()
 
             download_requests_stream(request, self.file)
+
+    def verify_checksum(source_checksum, checkfile):
```

Comment:
No, I didn't really know where to put it, so I put it in `FileBase`, but it makes much more sense in `sources/__init__.py`
Moved and modified the implementation to one similar to your example.